### PR TITLE
Add initial support for xstate/store/solid

### PR DESCRIPTION
**Pull request by steveadams**
I've added `useSelector` support to xstate/store for SolidJS which seems to be working well in my tests. Let me know what I can improve and I'd be happy to do it.

### CODE REVIEW

**Comment by steveadams** [comment]
Apologies for the force pushes; I can move this to a clean branch with a single commit instead if they're a problem.

File: packages/xstate-store/src/solid.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * ```
+ *
+ * @param store The store, created from `createStore(…)`
+ * @param selector A function which takes in the snapshot and returns a selected
+ *   value from it
+ * @param compare An optional function which compares the selected value to the
+ *   previously selected value
+ * @returns A read-only signal of the selected value
+ */
+export function useSelector<TStore extends Store<any, any>, T>(
```
Comment (steveadams) [inline packages/xstate-store/src/solid.ts:57]: I initially went with the name `createSelector`, and I might actually prefer it. I'm not sure what the consensus is on naming functions like this in SolidJS. The library itself has `use*` as well as `create*` naming conventions for functions like this.

Comment (davidkpiano) [reply packages/xstate-store/src/solid.ts:57]: That's fine, whatever is most idiomatic!

Comment (steveadams) [reply packages/xstate-store/src/solid.ts:57]: I'm just noticing the other solid-related code uses the `useActor` or `useMachine` convention so I guess this makes sense.

File: packages/xstate-store/src/solid.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/* @jsxImportSource solid-js */
+import { AnyActorRef } from 'xstate';
```
Comment (steveadams) [inline packages/xstate-store/src/solid.ts:2]: Whoops, will remove (left over from experiments)

**Comment by steveadams** [comment]
You're welcome! I really enjoyed this process and I'm stoked to contribute to XState.

**Pull request merged**